scipy/sparse/csgraph/tests/test_shortest_path.py

```diff
@@ -97,6 +97,34 @@ def check(method, directed_in):
             check(method, directed_in)
 
 
+def test_dijkstra_indices_min_only():
+
+    SP_res = {True: directed_SP,
+              False: undirected_SP}
+    index_list = [np.array([0, 2, 4], np.int64),
+                  np.array([0, 4], np.int64),
+                  np.array([3, 4])]
+
+    def check(directed, indices):
+        SP_ans = np.array(SP_res[directed])
+        min_ind_ans = indices[np.argmin(SP_ans[indices, :], axis=0)]
+        min_d_ans = np.zeros(SP_ans.shape[0], SP_ans.dtype)
+        for k in range(SP_ans.shape[0]):
+            min_d_ans[k] = SP_ans[min_ind_ans[k], k]
+        min_ind_ans[np.isinf(min_d_ans)] = -9999
+
+        SP, pred, sources = dijkstra(directed_G,
+                                     directed=directed,
+                                     indices=indices,
+                                     min_only=True,
+                                     return_predecessors=True)
+        assert_array_almost_equal(SP, min_d_ans)
+        assert_array_equal(min_ind_ans, sources)
+    for indices in index_list:
+        for directed in (True, False):
+            check(directed, indices)
```

Comment:
This formulation could be streamlined and rendered more readable as:
```
@pytest.mark.parametrize('directed, SP_ans',
                         ((True, directed_SP),
                          (False, undirected_SP))
@pytest.mark.parametrize('indices', ([0, 2, 4], [0, 4], [3, 4])
def test_dijkstra_indices_min_only(directed, SP_ans, indices):
    <contents of check function>
```

Reply:
You can. We are adopting `pytest` practices bit by bit. Might as well take advantage of the readability improvements they would provide in this PR

Reply:
I can leave it half done you mean?

Reply:
Yes. It's possible that we will (effectively) never bother to change the existing tests to `pytest` style, since they work fine and doing so makes things like `git blame` and history browsing harder anyway.

Reply:
push a commit with the parameterized tests